src/expire.c

```diff
@@ -521,6 +569,45 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
         return;
     }
 
+    /* NX option is set, check current expiry */
+    if (flag & FLAG_NX) {
+        current_expire = getExpire(c->db, key);
+        if (current_expire != -1) {
+            addReply(c,shared.czero);
```

Comment:
I still suggest not to use the return value of `EXPIRE` commands for TTL. It introduces additional complexity for various clients to distinguish *return code* and *value* for this write operation, that defined by an optional flag. 

And also I think returning `0` makes sense for this new scenario. We can extend its meaning along with these new features. The `0` means nothing changed with this write operation, it can be a key not exist or the operation is skipped.

Reply:
ok. we can stop here. already a big step forward.
GET and INCR arguments can be added later if we wish (nothing here that prevents that)